src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,409 @@
+  /**
+   * It is 'The bottom of CdkStickyRegion of current sticky header - the height
+   * of current header', which is used with '_containerStart' to judge whether the current header
+   * need to be stuck.
+   */
+  private _scrollFinish: number;
+  /** The width of the sticky-header when it is stuck. */
+  private _scrollingWidth: number;
+
+  private _onScrollSubscription: Subscription;
+
+  private _onTouchSubscription: Subscription;
+
+  private _onResizeSubscription: Subscription;
+
+  /**
+   * A list of string which can be set to sticky-header's style.position
+   * and make Sticky positioning work.
+   */
+  private _supportList: Array<string>;
+
+  constructor(_element: ElementRef,
+              scrollable: Scrollable,
+              @Optional() public parentReg: CdkStickyRegion) {
+    this.element = _element.nativeElement;
+    this.upperScrollableContainer = scrollable.getElementRef().nativeElement;
+
+    this.getSupportList();
+    this.setStrategyAccordingToCompatibility();
+  }
+
+  ngAfterViewInit(): void {
+    if(this.isIE === true) {
+      this.stickyParent = this.parentReg != null ?
+        this.parentReg._elementRef.nativeElement : this.element.parentElement;
+      this.originalCss = this.generateCssStyle(this.getCssValue(this.element, 'zIndex'),
+        this.getCssValue(this.element, 'position'), this.getCssValue(this.element, 'top'),
+        this.getCssValue(this.element, 'right'), this.getCssValue(this.element, 'left'),
+        this.getCssValue(this.element, 'bottom'), this.getCssValue(this.element, 'width'));
+      this.padding = this.getCssValue(this.element, 'padding');
+      this.stickyRegionHeight = this.getCssNumber(this.stickyParent, 'height');
+      this.attach();
+      this.defineRestrictionsAndStick();
+    }
+  }
+
+  ngOnDestroy(): void {
+    if (this._onScrollSubscription) {
+      this._onScrollSubscription.unsubscribe();
+    }
+
+    if (this._onResizeSubscription) {
+      this._onResizeSubscription.unsubscribe();
+    }
+
+    if (this._onTouchSubscription) {
+      this._onTouchSubscription.unsubscribe();
+    }
+  }
+
+  /**
+   * getSupportList() is used to get a list of string which can be set to
+   * sticky-header's style.position and make Sticky positioning work.
+   * It returns a list of string.
+   *
+   * According to the "Position:sticky Browser compatibility" in
+   * "https://developer.mozilla.org/en-US/docs/Web/CSS/position".
+   *
+   * For Desktop: Sticky positioning works well on Chrome, Edge, Firefox and Opera. And can
+   * also work well on Safari with a "-webkit-" prefix. It only does not work on IE.
+   *
+   * For Mobile: Sticky positioning works well on Android Webview, Chrome for Android, Edge,
+   * Firefox Mobile, Opera Mobile. And can also work well on Safari Mobile with a "-webkit-" prefix.
+   * It won't always work on IE phone.
+   *
+   */
+  getSupportList(): void {
+    let prefixTestList = ['', '-webkit-', '-ms-', '-moz-', '-o-'];
```

Comment:
Is this from Modernizer? Should we provide attribution and a link to the code in case we want to cross-reference it in the future?